From f9b5a5d85dfabdee590362b8c8f55e2fd1ad787d Mon Sep 17 00:00:00 2001
From: Andrew McDonnell <bugs@andrewmcdonnell.net>
Date: Sun, 3 May 2015 19:43:05 +0930
Subject: [PATCH] bmp085 at end of other.mk

---
 package/kernel/linux/modules/other.mk | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/package/kernel/linux/modules/other.mk b/package/kernel/linux/modules/other.mk
index 082c74a..1dfaf16 100644
--- a/package/kernel/linux/modules/other.mk
+++ b/package/kernel/linux/modules/other.mk
@@ -989,3 +989,23 @@ define KernelPackage/echo/description
 endef
 
 $(eval $(call KernelPackage,echo))
+
+
+define KernelPackage/bmp085-i2c
+  SUBMENU:=$(OTHER_MENU)
+  TITLE:=Bosch BMP085 and compatible I2C pressure sensor
+  DEPENDS:= +kmod-i2c-core +kmod-lib-lzo +kmod-regmap
+  KCONFIG:=CONFIG_BMP085_I2C \
+          CONFIG_REGMAP=y \
+          CONFIG_REGMAP_I2C=y
+  FILES:=$(LINUX_DIR)/drivers/misc/bmp085-i2c.ko
+  AUTOLOAD:=$(call AutoProbe,bmp085-i2c)
+endef
+
+define KernelPackage/bmp085-i2c/description
+ Kernel module for Bosch BMP085/BMP180 (and compatible) pressure sensor module
+ chips connected via I2C.
+endef
+
+$(eval $(call KernelPackage,bmp085-i2c))
+
-- 
1.9.1

